server/apps/users/services.py

```diff
@@ -90,6 +90,51 @@ def get_invisible_stages(user):
 
 class UsersService(BaseService):
 
+    def __is_invalid_operation(self, user, updates, method):
+        """
+        Checks if the requested 'PATCH' or 'DELETE' operation is Invalid.
+        Operation is invalid if one of the below is True:
+            1. Check if the user is updating his/her own status.
+            2. Check if the user is changing the status of other logged-in users.
+            3. A user without 'User Management' privilege is changing role/user_type/privileges
+
+        :return: error message if invalid.
+        """
+
+        if 'user' in flask.g:
```

Comment:
Agreed, but `auth.py` sticks user to `flask.g` and that's the reason we had to use `flask.g`